### PR TITLE
[Bug 1749312] metrics_clients_last_seen join on channel

### PR DESCRIPTION
**Pull request by scholtzan**
For reference: https://mozilla.slack.com/archives/C4D5ZA91B/p1641927084020700

It is possible that clients switch between channels (for example from nightly -> beta) and keep their `client_id`. This problem is currently affecting the most recent runs of `metrics_clients_last_seen` for Fenix:
https://workflow.telemetry.mozilla.org/log?dag_id=copy_deduplicate&task_id=metrics_clients_last_seen&execution_date=2022-01-08T01%3A00%3A00%2B00%3A00

We'll need to backfill `metrics_clients_last_seen` once this has been merged.

Checklist for reviewer:

- [ ] Commits should reference a bug or github issue, if relevant (if a bug is referenced, the pull request should include the bug number in the title)
- [ ] If the PR comes from a fork, trigger integration CI tests by running the [Push to upstream workflow](https://github.com/mozilla/bigquery-etl/actions/workflows/push-to-upstream.yml) and provide the `<username>:<branch>` of the fork as parameter. The parameter will also show up
in the logs of the `manual-trigger-required-for-fork` CI task together with more detailed instructions.
- [ ] If adding a new field to a query, ensure that the schema and dependent downstream schemas have been updated

For modifications to schemas in restricted namespaces (see [`CODEOWNERS`](/CODEOWNERS)):
- [ ] Follow the [change control procedure](https://docs.google.com/document/d/1TTJi4ht7NuzX6BPG_KTr6omaZg70cEpxe9xlpfnHj9k/edit#heading=h.ttegrcfy18ck)

### CODE REVIEW

File: sql_generators/glean_usage/templates/metrics_clients_last_seen.query.sql [hunk 1]
```diff
@@ -41,4 +41,5 @@ FULL JOIN
   _current
 USING
   (client_id,
-    sample_id)
+    sample_id,
+    normalized_channel)
```
Comment (jklukas) [inline sql_generators/glean_usage/templates/metrics_clients_last_seen.query.sql:45]: Are we guaranteed that `normalized_channel` is populated? IIRC, a row with a null in any of these join fields will get dropped.

**Review by jklukas — APPROVED** [review]
r+wc

File: sql_generators/glean_usage/templates/metrics_clients_last_seen.query.sql (hunk 1 above)
Comment (scholtzan) [reply sql_generators/glean_usage/templates/metrics_clients_last_seen.query.sql]: I don't think it's guaranteed. Though looking at some selected tables (Fenix) it does not seem to be `null`. But I guess we should handle the `null` cases.

File: sql_generators/glean_usage/templates/metrics_clients_last_seen.query.sql [hunk 2]
```diff
@@ -39,6 +39,10 @@ FROM
...
-USING
-  (client_id,
-    sample_id)
+ON
+  _previous.client_id = _current.client_id AND
+  _previous.sample_id = _current.sample_id AND
+  (
+    _previous.normalized_channel = _current.normalized_channel OR 
+    (_previous.normalized_channel IS NULL AND _current.normalized_channel IS NULL) 
+  )
```
Comment (scholtzan) [inline sql_generators/glean_usage/templates/metrics_clients_last_seen.query.sql:48]: @jklukas does this seem reasonable?

Comment (jklukas) [reply sql_generators/glean_usage/templates/metrics_clients_last_seen.query.sql:48]: This looks reasonable, though I'm not 100% sure without running it and validating the output. It's worth double-checking that we get the right result here before merging.

**Pull request merged**